spec/rubocop/cop/lint/ambiguous_operator_spec.rb

```diff
@@ -3,15 +3,90 @@
 RSpec.describe RuboCop::Cop::Lint::AmbiguousOperator do
   subject(:cop) { described_class.new }
 
+  context 'with `+` unary operator in the first argument' do
+    context 'without parentheses' do
+      context 'without whitespaces on the right of the operator' do
+        it 'registers an offense and corrects' do
+          expect_offense(<<~RUBY)
+            do_something(+24)
+            do_something +42
+                         ^ Ambiguous positive number operator. Parenthesize the method arguments if it's surely a positive number operator, or add a whitespace to the right of the `+` if it should be a addition.
+          RUBY
+
+          expect_correction(<<~RUBY)
+            do_something(+24)
+            do_something(+42)
+          RUBY
+        end
+      end
+
+      context 'with a whitespace on the right of the operator' do
+        it 'does not register an offense' do
+          expect_no_offenses(<<~RUBY)
+            do_something + 42
+          RUBY
+        end
+      end
+    end
+
+    context 'with parentheses around the operator' do
+      it 'does not register an offense' do
+        expect_no_offenses(<<~RUBY)
+          do_something(-42)
```

Comment:
Should be `+42` since the example is within the context "with `+` unary operator in the first argument".